src/librustdoc/clean/utils.rs

```diff
@@ -544,12 +544,14 @@ crate fn has_doc_flag(attrs: ty::Attributes<'_>, flag: Symbol) -> bool {
 }
 
 /// Return a channel suitable for using in a `doc.rust-lang.org/{channel}` format string.
+///
+/// NOTE: keep this in sync with `bootstrap::Builder::doc_rust_lang_org_channel`, or tests will fail on beta/stable.
 crate fn doc_rust_lang_org_channel() -> &'static str {
     match env!("CFG_RELEASE_CHANNEL") {
         "stable" => env!("CFG_RELEASE_NUM"),
```

Comment:
Hm, linking to CFG_RELEASE_NUM here seems pretty unfortunate - it implies we're going to end up with doc.rust-lang.org/1.45.0 being linked to itself, and will probably only make the "stale docs in google" problem worse?

Reply:
The reason I don't want to unconditionally link to /stable is because the docs could be different there - the section headings are not part of the stable API, and rustdoc/rustc occasionally delete entire pages of their books.

I haven't noticed stale docs being a problem in practice. The most discussion I've seen is on https://github.com/rust-lang/rust/issues/56700, which looks like it's been fixed by https://github.com/rust-lang/rust/pull/69992 ?

Reply:
That said, this is ultimately a @rust-lang/libs decision, happy to go with whatever you decide.

Reply:
> The reason I don't want to unconditionally link to /stable is because the docs could be different there - the section headings are not part of the stable API, and rustdoc/rustc occasionally delete entire pages of their books.

I would think that almost all docs pages that weren't built with a very recent release are using Docs.rs, and those will be using nightly, so they won't link to `/1.52.0`. So maybe linking to `/stable` wouldn't be that much of an issue? Also, linking to `/stable` when appropriate is at least better than unconditionally linking to `/nightly`.

Reply:
@camelid this is comparing linking /1.52 to linking to /stable; I don't think nightly is relevant.